npm/webpack-dev-server/src/loader.ts

```diff
@@ -47,20 +48,24 @@ function buildSpecs (projectRoot: string, files: Cypress.Cypress['spec'][] = [])
 
 // Runs the tests inside the iframe
 export default function loader () {
-  const { files, projectRoot } = this._cypress as { files: Cypress.Cypress['spec'][], projectRoot: string }
+  const { files, projectRoot, support } = this._cypress as CypressOptions
+
+  const supportFileAbsolutePath = JSON.stringify(path.resolve(projectRoot, support))
 
   return `
+  var loadSupport = ${support ? `() => import(${supportFileAbsolutePath})` : `() => Promise.resolve()`}
   var allTheSpecs = ${buildSpecs(projectRoot, files)};
 
-  const { init } = require(${JSON.stringify(require.resolve('./aut-runner'))})
+  var { init } = require(${JSON.stringify(require.resolve('./aut-runner'))})
 
-  const { restartRunner } = init(Object.keys(allTheSpecs)
-    .filter(key => allTheSpecs[key].shouldLoad())
-    .map(a => allTheSpecs[a].load())
-  )
+  var scriptLoaders = Object.values(allTheSpecs).reduce(
+    (accSpecLoaders, potentialSpecLoader) => {
+      if (potentialSpecLoader.shouldLoad()) {
+        accSpecLoaders.push(potentialSpecLoader.load())
+      }
+      return accSpecLoaders
+  }, [loadSupport()])
```

Comment:
Why are we invoking .load inside of the loader instead of the driver? Didn't we just say that invokes the promise early?

Stylistically, I think shorter variable names are important for readability in this case

```suggestion
  const scriptLoaders = Object.values(allTheSpecs).reduce(
    (acc, spec) => {
      if (spec.shouldLoad()) {
        acc.push(spec.load)
      }
      return acc
  }, [loadSupportFile])
```